### PR TITLE
Add new test flow and go-test job

### PR DESCRIPTION
**Pull request by ashmckenzie**
This PR aims to bring in the necessary logic in order to run `go test ./...` against the codebase.

### CODE REVIEW

**Comment by ashmckenzie** [comment]
@josegonzalez WDYT? 🙂

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    branches:
+      - 'master'
+
+jobs:
+  go-test:
+    name: go-test
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        go-version: ['1.22.0']
```
Comment (josegonzalez) [inline .github/workflows/test.yml:19]: Can you modify this to just use the go version specified in the go.mod?

Comment (ashmckenzie) [reply .github/workflows/test.yml]: Updated 👍

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+jobs:
+  go-test:
+    name: go-test
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
```
Comment (josegonzalez) [inline .github/workflows/test.yml:22]: Maybe this instead? https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

Comment (ashmckenzie) [reply .github/workflows/test.yml]: Good idea, updated 🏓

**Pull request merged**